app/models/parameter.rb

```diff
@@ -7,7 +7,7 @@ class Parameter < ActiveRecord::Base
 
   belongs_to_host :foreign_key => :reference_id
   include Authorizable
-  validates :name, :presence => true, :no_whitespace => true
+  validates :name, :presence => true, :no_whitespace => true, :uniqueness => { :scope => :reference_id }
```

Comment:
Would uniqueness scope include :type as well?  Putting it in the subclass would do this automatically.